kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/operator/KroxyliciousOperatorBundleInstaller.java

```diff
@@ -0,0 +1,272 @@
+
+        return Objects.equals(kroxyliciousOperatorName, otherInstallation.kroxyliciousOperatorName) &&
+                Objects.equals(namespaceInstallTo, otherInstallation.namespaceInstallTo) &&
+                Objects.equals(extraLabels, otherInstallation.extraLabels);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(cluster, extensionContext, kroxyliciousOperatorName, namespaceInstallTo, extraLabels);
+    }
+
+    @Override
+    public String toString() {
+        return "KroxyliciousOperatorBundleInstaller{" +
+                "cluster=" + KroxyliciousOperatorBundleInstaller.cluster +
+                ", extensionContext=" + extensionContext +
+                ", kroxyliciousOperatorName='" + kroxyliciousOperatorName + '\'' +
+                ", namespaceInstallTo='" + namespaceInstallTo + '\'' +
+                ", extraLabels=" + extraLabels +
+                ", testClassName='" + testClassName + '\'' +
+                ", testMethodName='" + testMethodName + '\'' +
+                '}';
+    }
+
+    private void setTestClassNameAndTestMethodName() {
+        this.testClassName = this.extensionContext.getRequiredTestClass() != null ? this.extensionContext.getRequiredTestClass().getName() : "";
+
+        try {
+            if (this.extensionContext.getRequiredTestMethod() != null) {
+                this.testMethodName = this.extensionContext.getRequiredTestMethod().getName();
+            }
+        }
+        catch (PreconditionViolationException e) {
+            LOGGER.debug("Test method is not present: {}\n{}", e.getMessage(), e.getCause());
+            // getRequiredTestMethod() is not present, in @BeforeAll scope so we're avoiding PreconditionViolationException exception
+            this.testMethodName = "";
+        }
+    }
+
+    @Override
+    public void install() {
+        LOGGER.info("Install Kroxylicious Operator via Yaml bundle in namespace {}", namespaceInstallTo);
+
+        setTestClassNameAndTestMethodName();
+
+        prepareEnvForOperator(namespaceInstallTo);
+    }
+
+    @Override
+    public synchronized void delete() {
+        LOGGER.info(SEPARATOR);
+        if (IS_EMPTY.test(this)) {
+            LOGGER.info("Skip un-installation of the Kroxylicious Operator");
+        }
+        else {
+            LOGGER.info("Un-installing Kroxylicious Operator from Namespace: {}", namespaceInstallTo);
+
+            // clear all resources related to the extension context
+            try {
+                if (!Environment.SKIP_TEARDOWN) {
+                    for (File operatorFile : getFilteredOperatorFiles(installFiles)) {
+                        kubeClient().getClient().load(new FileInputStream(operatorFile.getAbsolutePath())).inAnyNamespace().delete();
+                    }
+                    ResourceManager.getInstance().deleteResources();
+                }
+            }
+            catch (Exception e) {
+                Thread.currentThread().interrupt();
```

Comment:
Why interrupt the thread?  You'd normally only do this in response to a InterruptedException